### PR TITLE
Namespacing with caseless enums

### PR DESCRIPTION
**Pull request by FranzBusch**
This is the follow-up PR regarding our discussion in the previous guild:

I still think we should follow recommended practices by the Swift team here. Furthermore, I looked at the Raywenderlich and Google Swift coding conventions and both of them use caseless enums for namespacing. I will attach a couple of links where arguments about this topic are made. SwiftLint also provides a rule which can enforce this convention. Until there is no `namespace` keyword in Swift this is the best solution in my opinion.

https://google.github.io/swift/
https://github.com/raywenderlich/swift-style-guide#constants
https://www.natashatherobot.com/swift-enum-no-cases/
https://ericasadun.com/2016/07/18/dear-erica-no-case-enums/
https://forums.swift.org/t/namespace-keyword/11338/2

### CODE REVIEW

**Comment by camina-apps** [comment]
there is one main difference all the links are talking about static constants not static methods 
```
enum Math {
  static let e = 2.718281828459045235360287
  static let root2 = 1.41421356237309504880168872
}
```

File: README.md [hunk 1]
```diff
@@ -267,4 +267,27 @@ instead of:
 1. Keep imports minimal. For example, don't import UIKit when importing Foundation will suffice.
 2. When introducing a new import, check if that is already importing an existent framework in your file. For instance, UIKit is already importing Foundation. Therefore, if you need UIKit, you can remove your previous Foundation import.
 
+## Namespacing
```
Comment (iSapozhnik) [inline README.md]: Please, include it to the Table of contents

File: README.md [hunk 2]
```diff
@@ -267,4 +267,27 @@ instead of:
...
+Declaring an enum without cases is the canonical way to define a “namespace” to group a set of related declarations, such as constants or helper functions. This enum automatically has no instances and does not require that extra boilerplate code be written to prevent instantiation.
+
+```swift
+enum Dimensions {
```
Comment (iSapozhnik) [inline README.md]: So the rule is to group variables only? Coz usually namespacing includes everything including methods.

**Comment by FranzBusch** [comment]
@camina-apps  At least of the two linked articles have examples for methods as well, I will update the PR to include them here too. [Here](https://forums.swift.org/t/submodule-is-an-ambiguous-term/12484) is another post from the Swift forums where one of the compiler engineers gives an example including a static method. In general, the rule is the same for methods as for constants, i.e. stuff that would otherwise end up in the global namespace or where we want to make a semantic grouping through a common namespace should be namespaced by using `enum`. In cases where the static methods belong to a regular struct or class with properties etc. nothing changes.

@iSapozhnik No my examples were just badly chosen, the namespacing rule should apply to constants as well as methods.

**Pull request merged**